src/Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -1,47 +1,140 @@
+        }}
+    }}
+
+";
+            stringBuilder.Append(constructor);
+            prettyFormatter.Append(constructor);
 
+            foreach (var column in dataFrame.Columns)
+            {
+                string columnName = column.Name.Replace(" ", string.Empty);
+                Type dataType = column.DataType;
+                string typeName = GetFriendlyName(column.GetType());
+                stringBuilder.Append($@"
+    public {typeName} {columnName} 
+    {{
+        get
+        {{
+            int columnIndex = Columns.IndexOf(""{columnName}"");
+            return Columns[columnIndex] as {typeName};
+        }}
+    }}");
+                stringBuilder.AppendLine();
+                prettyFormatter.Append($@"
+    public {typeName} {columnName} {{ get; }}");
+                prettyFormatter.AppendLine();
+            }
+
+            stringBuilder.AppendLine();
+            stringBuilder.Append(@"}");
+            prettyFormatter.AppendLine();
+            prettyFormatter.Append(@"}");
+            prettyFormatter.AppendLine();
+            return stringBuilder.ToString();
+        }
+
+        public Task OnLoadAsync(IKernel kernel)
+        {
             var kernelBase = kernel as KernelBase;
-            var directive = new Command("#!doit")
+            var directive = new Command("#!AddStrongTypes")
             {
-                Handler = CommandHandler.Create(async (FileInfo csv, string typeName, KernelInvocationContext context) =>
+                Handler = CommandHandler.Create(async (string dataFrameName, string typeName, KernelInvocationContext context) =>
                 {
                     // do the job
-                    var command = new SubmitCode(@$"public class {typeName}{{}}");
-                    context.Publish(new DisplayedValueProduced($"emitting {typeName} from {csv.FullName}", context.Command));
-                    await context.HandlingKernel.SendAsync(command);
+                    try
+                    {
+                        if (context.HandlingKernel is CSharpKernel cSharp)
+                        {
+                            System.Collections.Immutable.ImmutableArray<CodeAnalysis.Scripting.ScriptVariable> variables = cSharp.ScriptState.Variables;
+                            for (int i = 0; i < variables.Length; i++)
+                            {
+                                CodeAnalysis.Scripting.ScriptVariable variable = variables[i];
+                                if ((dataFrameName == null || variable.Name == dataFrameName) && variable.Value is DataFrame df)
+                                {
+                                    string typedDataFrame = GetTypedDataFrameWithProperties(df, typeName, out StringBuilder prettyFormatter);
+                                    context.Publish(new DisplayedValueProduced($"Created type {typeName}: ", context.Command));
+                                    await context.DisplayAsync(prettyFormatter.ToString());
```

Comment:
Why are we displaying different code than what we are generating/compiling?

Reply:
I thought this looked better since we're only seeing the API rather than spitting out the code being compiled. Would you rather see the implementation too? I tried that first and it looked rather noisy. I would think most folks aren't interested in looking at the implementation?

Reply:
My thinking is that someone could use this tool to generate code that they can use in their project and use it in the library/application. Not just in a notebook.

Outputting the generated code could be an option that is displayed by default.